packages/adapters/src/kubernetes-instance-adapter.ts

```diff
@@ -0,0 +1,178 @@
+/* eslint-disable no-console */
+import {
+    ExitCode,
+    IComponent,
+    ILifeCycleAdapterMain,
+    ILifeCycleAdapterRun,
+    IObjectLogger,
+    K8SAdapterConfiguration,
+    MonitoringMessageData,
+    SequenceConfig,
+    STHConfiguration,
+} from "@scramjet/types";
+
+import path from "path";
+import { ObjLogger } from "@scramjet/obj-logger";
+import { createReadStream } from "fs";
+import { KubernetesClientAdapter } from "./kubernetes-client-adapter";
+
+import { JsonDecoder } from "ts.data.json";
+
+const adapterConfigDecoder = JsonDecoder.object<K8SAdapterConfiguration>({
```

Comment:
Nice typeguard. We should have this in the config. Let's create a backlog entry out of this.

Reply:
https://github.com/scramjetorg/transform-hub/issues/279